### PR TITLE
OCP4: Add assessment for section 1 for PCI-DSS

### PR DESCRIPTION
**Pull request by JAORMX**
This adds the assessment for PCI-DSS's section 1 and fills in the rules
tthat are applicable that are already written.

Rules that are yet to be written will be added in subsequent pull
requests.

Signed-off-by: Juan Antonio Osorio Robles <jaosorior@redhat.com>

### CODE REVIEW

File: controls/pcidss_ocp4.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 policy: PCI-DSS
 title: Configuration Recommendations of a GNU/Linux System
-id: pcidss
+id: pcidss_ocp4
```
Comment (jhrozek) [inline controls/pcidss_ocp4.yml:3]: Does renaming the ID from `pcidss` have any impact on the references in the rules that still use `pcidss` ?

File: controls/pcidss_ocp4.yml [hunk 2]
```diff
@@ -35,7 +37,10 @@ controls:
       cardholder data environment and other networks, including any wireless networks
     levels:
     - base
-    notes: ''
+    notes: |-
+      While there are ways and tools where the OpenShift Container Platform could
+      help address this control, covering the scope of a deployment with the application
+      specific connections is out of scope.
```
Comment (jhrozek) [inline controls/pcidss_ocp4.yml:43]: Would it make sense to recommend stackrox/ACS here?

File: controls/pcidss_ocp4.yml [hunk 3]
```diff
@@ -44,7 +49,10 @@ controls:
       and networks
     levels:
     - base
-    notes: ''
+    notes: |-
+      While there are ways and tools where the OpenShift Container Platform could
+      help address this control, covering the scope of a deployment with the application
+      specific connections is out of scope.
```
Comment (jhrozek) [inline controls/pcidss_ocp4.yml:55]: Same q. about Stackrox/ACS

File: controls/pcidss_ocp4.yml [hunk 4]
```diff
@@ -19,14 +19,16 @@ controls:
   levels:
   - base
   notes: ''
-  status: not applicable
+  status: automated
```
Comment (jhrozek) [inline controls/pcidss_ocp4.yml:22]: Can you explain to me the relation between the status in the section header and the individual rules? In this section, it seems we have most not applicable and one automated, but in section 1.3, we have some automated rules, but the status of the section 1.3 as a whole is 'pending'.

File: controls/pcidss_ocp4.yml [hunk 5]
```diff
@@ -144,68 +181,95 @@ controls:
...
     - base
-    notes: ''
-    status: pending
-    rules: []
+    notes: |-
+      To a limited degree, OpenShift's ingress routing (enabled by default)
+      and NetworkPolicies can support restriction of inbound traffic to pods.
+      Further restricting needs to happen on the cloud where OpenShift is deployed
+      at, but that's outside of OpenShift's scope.
+    status: automated
```
Comment (jhrozek) [inline controls/pcidss_ocp4.yml:189]: Should this be automated or partial?

File: controls/pcidss_ocp4.yml [hunk 6]
```diff
@@ -144,68 +181,95 @@ controls:
...
     - base
-    notes: ''
-    status: pending
+    notes: |-
+      OpenShift can support authorization of outbound traffic via
+      egress  policy  objects[1][2].
+
+      [1] https://docs.openshift.com/container-platform/4.6/networking/openshift_sdn/configuring-egress-firewall.html
+      [2] https://docs.openshift.com/container-platform/4.6/networking/ovn_kubernetes_network_provider/configuring-egress-firewall-ovn.html
+    status: supported
```
Comment (jhrozek) [inline controls/pcidss_ocp4.yml:232]: Should we have a CaC rule for this control?

**Review by jhrozek — COMMENTED** [review]
Nice, I left a couple of comments, mostly down to me being unfamiliar with the controls format in CaC..

File: controls/pcidss_ocp4.yml [hunk 7]
```diff
@@ -144,68 +181,95 @@ controls:
...
+      * Egress Policies
+
+      Special attention needs to be taken to audit that these components
+      are appropriately segregated by the aforementioned mechanisms
 
-  - id: Req-1.3.8
-    title: 1.3.8 Do not disclose private IP addresses and routing information to unauthorized
-      parties.
+      [1] https://docs.openshift.com/container-platform/4.8/networking/network_policy/about-network-policy.html
+      [2] https://docs.openshift.com/container-platform/4.6/networking/openshift_sdn/configuring-egress-firewall.html
```
Comment (mrogers950) [inline controls/pcidss_ocp4.yml:262]: Should we point to 4.8 docs for the rest of these? (or use the /latest path)

File: controls/pcidss_ocp4.yml [hunk 8]
```diff
@@ -231,17 +297,23 @@ controls:
       employee-owned devices.
   levels:
   - base
-  notes: ''
-  status: pending
+  notes: |-
+    personal firewall software is outside the scope of the OpenShift
```
Comment (mrogers950) [inline controls/pcidss_ocp4.yml:301]: capitalize Personal

File: controls/pcidss_ocp4.yml [hunk 9]
```diff
@@ -231,17 +297,23 @@ controls:
...
   title: 1.5 Ensure that security policies and operational procedures for managing
     firewalls are documented, in use, and known to all affected parties.
   levels:
   - base
-  notes: ''
-  status: pending
+  notes: |-
+    The organization will want to ensure that they include in security
+    policies and operational procedures for managing  the  OpenShift
+    SDN  and  OpenShift  NetworkPolicy  objects  within  the  OpenShift
```
Comment (mrogers950) [inline controls/pcidss_ocp4.yml:314]: There' some double spacing spread around, maybe unintentional

File: controls/pcidss_ocp4.yml (hunk 1 above)
Comment (JAORMX) [reply controls/pcidss_ocp4.yml:3]: This is the ID of the controls and not the benchmark. It doesn't affect references

File: controls/pcidss_ocp4.yml (hunk 6 above)
Comment (JAORMX) [reply controls/pcidss_ocp4.yml:232]: yes. We should; that'll come in a separate PR... at some point

File: controls/pcidss_ocp4.yml (hunk 4 above)
Comment (JAORMX) [reply controls/pcidss_ocp4.yml:22]: When I proposed the status I actually didn't think about control groups. Given that only one of the sub-controls belonging to this group was automatable and is implemented, I decided to mark this whole group as `automated`. ideas or alternatives are welcome.

File: controls/pcidss_ocp4.yml (hunk 2 above)
Comment (JAORMX) [reply controls/pcidss_ocp4.yml:43]: done

File: controls/pcidss_ocp4.yml (hunk 3 above)
Comment (JAORMX) [reply controls/pcidss_ocp4.yml:60]: done

File: controls/pcidss_ocp4.yml (hunk 7 above)
Comment (JAORMX) [reply controls/pcidss_ocp4.yml]: done

File: controls/pcidss_ocp4.yml (hunk 9 above)
Comment (JAORMX) [reply controls/pcidss_ocp4.yml]: done

**Review by JAORMX — COMMENTED** [review]
Thanks for the feedback. Addressed the comments.

**Comment by JAORMX** [comment]
Pull-request updated, HEAD is now 69b6ac58baf64b67a0e9eb86792e2264999403cf

File: controls/pcidss_ocp4.yml (hunk 4 above)
Comment (jhrozek) [reply controls/pcidss_ocp4.yml:22]: OK,  this works as long as we are consistent

File: controls/pcidss_ocp4.yml (hunk 6 above)
Comment (jhrozek) [reply controls/pcidss_ocp4.yml:244]: OK

**Review by jhrozek — APPROVED** [review]
/lgtm
all my changes were done

**Pull request merged**